DM-35050: Revise Transinet technote with new design (#3)
lsst-dm/dmtn-216: body.tex

\section{Problem Definition -- Big Picture}
\label{sec:definition}

> Loosely speaking, the high-level objective in transient-detection is to locate intrinsically(?)-varying objects in each recently shot image of the sky.

Reviewer: ```suggestion
Loosely speaking, the high-level objective in transient-detection is to locate intrinsically-varying objects in each recent image of the sky.
```

\end{figure}

> The \emph{thresholding} step has traditionally been implemented as a simple $5\sigma$-thresholding, but other auxiliary approaches, such as pre-convolution with the PSF, may wrap around this stage to make it smart -- figure~\ref{fig:diagram}

Reviewer: I think the phrase "smart" here isn't meaningful (and hence isn't helpful).  The statistical justifications for detection in this way are described (for example) [here](https://dmtn-015.lsst.io/#kaiser-coadds) and [here](https://dmtn-015.lsst.io/#detection-maps); the arguments are very similar for image differences.

\item starting off $(I_t,I_s)$ -- a.k.a end-to-end, TransiNet-style.
\end{itemize}

> Each of the approaches has its own cons and pros, and methods based on both will be implemented for LSST. In this tech-note we focus on details of the former, the real/bogus classifier, and the end-to-end TransiNet will be covered in a future note.

Reviewer: again, a few citations to prior work would be appropriate here.

\section{Basic Real/Bogus Classifier}
\label{sec:basic}

\begin{figure}[h]
  \centering
  \includegraphics[width=.6\textwidth]{material/rb-classifier}
  \includegraphics[width=.6\textwidth]{material/rb-classifier-mod}

>   \caption{Classical Real/bogus classifier (top, \cref{sec:basic}) and the modified 3D classifier (bottom,  \cref{sec:modified})}

Reviewer: Citations to the "classical" example would be useful--e.g., the braai classifier uses all three images for input.

\end{equation}

where \emph{det} is a flag representing whether a source is detected by the upstream modules or not, and is usually implemented as a $5\sigma$ thresholding of the output of the image differencing module -- see \cref{fig:diagram}. Note that since typical neural networks do not estimate probability distributions, we do not use a probabilistic notation in \cref{eq:conditional}.

% Although the output of a classifier is categorical (i.e. a class label), we will be using the output before the last Softmax~\footnote{TBD} layer, which can be interpreted as a probability or score.

\subsection{The score}
As the task definition embraces a recognition according to spatial \emph{and} temporal features simultaneously, the score needs to be defined carefully to capture both aspects.
More specifically, we need to define the term ``real'' more precisely: a non-astrophysical detection, such as an artifact, is obviously not ``real''. But how about a non-varying astrophysical source?

\begin{figure}[h]
  \centering
  \includegraphics[width=.8\textwidth]{material/score-space}
  \caption{The 2-dimensional realness space}
  \label{fig:score-space}
\end{figure}

> As illustrated in figure~\ref{fig:score-space}, the real-ness space is composed of (at least) two dimensions, which we name ``transient-ness'' and ``object-ness''.

Reviewer: While I understand what you mean here, I suspect if we could come up with other names it would be less confusing--"transient" and "object" already having technical definitions in LSST...

Author: I'm gonna need some help here, as I can't seem to find much better alternatives.
As for the term "transient", I think here it refers to the same concept as in LSST in general. Maybe something like "variability"? (but then again the conflict gets deeper!)

I think the term "object" is also too popular to be booked for a very narrow definition. Here I am trying to refer to astrophysical objects as opposed to artifacts. I can think of "real-ness", but we have already used that for something else :)

Any suggestions are appreciated :)

Reviewer: "transient-ness" I think is a roundabout way of saying "signal to noise ratio"

"object-ness" is effectively 1-spuriousness, which we already use for this purpose in LSST.

\end{tcolorbox}

This way, in the training phase we inject our prior knowledge about the ``real-ness'' of each object into the learning process, regardless of the difficulty of the detection of each object and without adding artificial Aleatoric uncertainty to the system. Later during the test phase (i.e. normal working mode of the pipeline), we use the pseudo-probabilities appearing before thresholding at the final layer of the network as a proxy of the score, which capture the Epistemic uncertainty of the model too.

% There are multiple available options:
% \begin{itemize}
% \item simple binary classification -- the resulting score may not be quite informative
% \item regression -- more difficult to define
% \item multi-class classification -- difficult data annotation
% \end{itemize}

% If we go with regression... there are bogus detections, variables, ``appearing transients''
% \begin{equation}
%   s=\left| \frac{A_2-A_1}{A_2+A_1} \right|
% \end{equation}

\subsection{Drawbacks}
\begin{itemize}

> \item has no access to the temporal~\footnote{throughout this document the term ``temporal'' is used to represent the transition from a template/reference image to a science image. It has to do, but should not be confused with the ``real'' time axis!} behavior of the source. This is by definition outsourced to the upstream image differencing module and hence there is no ``learned intelligence'' in processing the ``transient-ness'' of the candidate.

Reviewer: With a classifier taking three stamps, though, this isn't true: aside from false negatives (which you cover in the next bullet) the classifier has no disadvantage relative to the image differencing algorithm.

If we can avoid the "temporal" language here, again I think other readers will be less confused.

Author: Mostly agreed -- there is the crop-based-on-diff concept too. I think the explanation in Section 4 covers this in an elaborate fashion. I'm adding a pointer to that section too.

Author: Oh, do you have any suggestions for a replacement for the term "temporal"? 
Actually, I like it a lot, as it fits well with "spatial", and together they form the two main dimensions in which we seek information...

Reviewer: Again, I think the fact that you can do classical RB with three stamps makes this discussion moot.

\subsection{Benefits}
\label{sec:rb_benefits}

Although as discussed this is just a sub-optimal learning-based approach to this problem, it still has the advantage of \emph{interpretability}; the generated output might be too contaminated with false alarms and (at the same time) missed true transients, but the process is fully transparent to the downstream users of the alerts and they have the possibility to develop their own post-processing blocks per need.

\subsection{What will be learned?}
Since the temporal behavior of the candidates fed to this module is assumed, it is foreseeable that the neural network will eventually learn a concept close to ``PSF-ness'', true to its title; real/bogus classification.

\section{Modified 3D Real/Bogus Classifier}
\label{sec:modified}
In this implementation, we try to feed the network with a glimpse of the temporal behaviour of the candidate, by passing the template and science image pairs along with the diff image. This allows the network to make use of the temporal behavior of the source too.

> Note however that still the difference image plays a key role in the process, as the images are cropped around the sources detected in the difference image. This means that the modification only aims for clealiness of the detections, by passing more information about the existing candidates (diaSources), but cannot affect the completeness of detections, by definition.

Reviewer: ```suggestion
Note however that still the difference image plays a key role in the process, as the images are cropped around the sources detected in the difference image. This means that the modification only aims for cleanliness of the detections, by passing more information about the existing candidates (diaSources), but cannot affect the completeness of detections, by definition.
```

\section{Future Work}

> \subsection{End-to-End Simultaneous Localization and Classification (TransiNet)}

Reviewer: ```suggestion
\subsection{End-to-End Simultaneous Identification and Localization (TransiNet)}
```

"Classification" has existing other meanings to science users so it would be useful to avoid it.

Author: This is the terminology conflict (between Vision and Astro) where we should be very careful not to be wrong in either of the areas. What does "Classification" exactly mean in astro? 
Or put differently, what is a solution to "real transient or not?" called in astro?

Reviewer: "Classification" in this context in astro is typically meant to mean determining what the astrophysical origin of the emission is (e.g., a supernova, an AGN, etc.).

maybe "simultaneous identification and artifact filtering"?  it's not perfect though.
